.github/CONTRIBUTING.md

```diff
@@ -0,0 +1,89 @@
+# How to contribute
+Thank you for showing interest in contributing to Pretendo Network! As an open source project, Pretendo Network welcomes contributions from developers outside of our core team. We are a large project, made up of over 100 different repositories spanning various languages and scopes. For the sake of consistency and maintainability, we have created these guidelines to outline project goals, standards, and best practices. This document will cover both code contributions as well as going over how to report any bugs or issues with a service or documentation. By adhering to these guidelines, contributors can ensure that their contributions align with the project's vision and meet the quality standards expected by the community.
+
+### Contents
+- [Prerequisites](#prerequisites)
+- [Making Issues](#making-issues)
+- [Code Changes](#code-changes)
+	- [Consistency](#consistency)
+	- [Quality and Substance](#quality-and-substance)
+- [Commits](#commits)
+	- [Scale and Scope](#scale-and-scope)
+	- [Messages](#messages)
+- [Pull Requests](#pull-requests)
+
+# Prerequisites
+Before diving into the contribution process, we recommend familiarizing yourself with our project's existing repositories. By studying our existing codebases, issues, and pull requests you will get a better sense of direction of how we operate. Additionally, please review our [Code of Conduct](CODE_OF_CONDUCT.md) to ensure a respectful and inclusive environment for all participants.
+
+Given that our project's main goal is to emulate Nintendo Network, it is also highly recommended that you familiarize yourself with how these services operate. Information on these topics, and our libraries, can be found both on our [wiki](https://nintendo-wiki.pretendo.network/docs/) and our (in progress) [developer documentation](https://developer.pretendo.network/home).
+
+> [!NOTE]
+> Developer documentation is highly work in progress. Do not expect to find everything there at this time. Please refer to the wiki first and foremost.
+
+Finally, we recommend joining our [Discord server](https://invite.gg/pretendo) to see any roadmap plans or ask specific questions from our developers.
```

Comment:
```suggestion
Finally, we recommend joining our [Discord server](https://invite.gg/pretendo) to see any roadmap plans or ask specific questions from our developers. To ask a technical question, please create a thread in the [#support channel](https://discord.com/channels/408718485913468928/1067162922989662360) and @mention a relevant developer who has recently committed to the repository in question. If you don't know who to mention or if it is a general question about multiple servers, mention @jonbarrow.
```

Reply:
Might need to clarify which questions are more appropriate for a GitHub issue using the question template vs a discord question. Or just the user's choice.

Reply:
> could this be accepted then updated if there is a change to the public channels/server?

The idea itself is pretty solid but we haven't settled on the best way to go about this yet, which is why this hasn't been accepted. The support channel is *intended* for people with general service issues, not necessarily technical discussions. I'm not sure that trying to reuse that channel specifically would do much good, as it would just start to clog the channel up a bit and wouldn't really solve the issue of getting developer attention (this was mentioned on Discord yesterday, but we tend to opt for a "community-first" style of support right now due to lack of staff. The community helps each other in those threads, and we only jump in when need be. Many of us don't even watch those threads ourselves, so any technical questions would have the same level of lacking visibility)

Honestly one of the biggest criticisms we get is people feeling like we rely on our Discord server for far too much (people dislike that it's one of the main ways to get support, since not everyone uses Discord). I've been thinking for a while now about just moving a lot of this out of Discord and into something like an online forum that we'd run. Moving stuff to our own forum on our website would solve several issues:

- Support outside of Discord
- Appeals for things like bans
- More organized player reports
- Asking technical questions

We can easily split these topics into their own categories on a forum to keep things organized, which would help with visibility a lot I think. I haven't brought this idea up with the rest of the team, but now is probably a good time

@DaniElectra @shutterbug2000 @SuperMarioDaBom @hauntii @gitlimes @ashquarky @InternalLoss @TraceEntertains @MatthewL246

Reply:
> The support channel is intended for people with general service issues, not necessarily technical discussions.

It probably does make sense to keep these things separate. Let the community continue to help people who need more basic help with getting things set up or fixing errors, and keep the technical discussion in a place with a lower volume of posts so developers can keep track of them without needing to ping people.

> moving a lot of this out of Discord and into something like an online forum that we'd run.

A forum sounds like a good idea to me. To be completely honest, I have no experience in contributing to online forums, but I really appreciate how they often provide great information about niche topics because they're public and indexed.

I'm conflicted about how organizing should work: I think that there are a lot of people on Discord who would be unwilling to sign up for an account on a forum, so it makes sense to keep the support channel available for them. (Not sure if any forum software has a "Log in with Discord" feature?) But the additional friction for joining a forum might help keep technical discussions higher quality? However, that also partially defeats the purpose of a forum, as people who don't use Discord will still be left in the dark on potentially useful information.

Reply:
> unwilling to sign up for an account on a forum

There's no need, we have that solved already; the PNID. This isn't my first time doing this, actually. Around 10 years ago I was the lead of a massive vanilla MMORPG Minecraft server, and we had our own custom forum that I wrote which used users Minecraft accounts as their forum accounts

Reply:
For some reason I didn't think about that, makes total sense!